### PR TITLE
feat(Select): add validation status and message

### PR DESCRIPTION
**Pull request by NatalieLoyzaga**
## Why

- Add Validation props to Select

## What

- Add `status` and `validationMessage` props
- Create validation story
image

### CODE REVIEW

File: packages/select/src/Select/Select.tsx [hunk 1]
```diff
@@ -39,6 +39,14 @@ export interface SelectProps
...
   ) => React.ReactNode
   children?: (optionsProps: SelectOptionsProps) => React.ReactNode
+  /**
+   * Updates the styling of the validation FieldMessage.
+   */
+  status?: "error" | "caution"
+  /**
+   * A descriptive message for the 'status' states.
+   */
+  validationMessage?: React.ReactNode | undefined
```
Comment (gyfchong) [inline packages/select/src/Select/Select.tsx:49]: Should we make the validation message required if there's a status? There doesn't seem to be a default 🤔

Comment (NatalieLoyzaga) [reply packages/select/src/Select/Select.tsx:49]: `type ValidationProps =
  | {
      status: "error" | "caution"
      validationMessage: React.ReactNode | undefined
    }
  | {
      status?: never
      validationMessage?: never
    }`

Comment (NatalieLoyzaga) [reply packages/select/src/Select/Select.tsx:49]:
```
type ValidationProps =
  | {
      status: "error" | "caution"
      validationMessage: React.ReactNode | undefined
    }
  | {
      status?: never
      validationMessage?: never
    }
    
    ```

Comment (NatalieLoyzaga) [reply packages/select/src/Select/Select.tsx:49]: I could do this and then `export const Select: React.FC<SelectProps & ValidationProps>`

Comment (NatalieLoyzaga) [reply packages/select/src/Select/Select.tsx:49]: thoughts @gyfchong ? Means I can only have status + validation message or niether

Comment (gyfchong) [reply packages/select/src/Select/Select.tsx:49]: Hmm I just had a look at the others, doesn't seem like we add any restriction on this, so nevermind in this case. I'll pose the question for a more universal change 👍

**Review by gyfchong — DISMISSED** [review]

File: packages/select/src/Select/Select.tsx [hunk 2]
```diff
@@ -90,6 +111,8 @@ export const Select: React.FC<SelectProps> & SubComponentProps = ({
     }
   }, [state.isOpen])
 
+  const shouldShowValidationMessage = !isDisabled && validationMessage
```
Comment (gyfchong) [inline packages/select/src/Select/Select.tsx:114]: Oh, according to design the message should show regardless of disabled state

Comment (NatalieLoyzaga) [reply packages/select/src/Select/Select.tsx:114]: I think this maybe a mistake but I will ask Elliot

Comment (NatalieLoyzaga) [reply packages/select/src/Select/Select.tsx:114]: wait what message are you seeing in the design?

The description message will show regardless the validation won't show if disabled

Comment (gyfchong) [reply packages/select/src/Select/Select.tsx:114]: image

I assume this condition will prevent the 4th column from happening?

**Review by gyfchong — APPROVED** [review]

File: packages/select/src/Select/Select.tsx (hunk 2 above)
Comment (dougmacknz) [reply packages/select/src/Select/Select.tsx]: This is something @HeartSquared and I were just talking about. The date picker has this opinion built in, which I'm not a fan of. I reckon we should leave this up to consumers.

Comment (HeartSquared) [reply packages/select/src/Select/Select.tsx]: This is a strange thing tbh. If it's disabled, does it even make sense to have any validation message on it? I feel like I remember having a chat with Jason about it earlier when we were working on the DatePicker and we figured it made no sense so we put the conditions in.
But yeah, I guess maybe we should just let them do it; weird as it is. Perhaps worth revisiting the whole validation structure pattern in general.

**Pull request merged**